cdap-securestore-ext-gcp-secretstore/src/main/java/io/cdap/cdap/securestore/gcp/cloudsecretmanager/WrappedSecret.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.securestore.gcp.cloudsecretmanager;
+
+import com.google.cloud.secretmanager.v1.Replication;
+import com.google.cloud.secretmanager.v1.Replication.Automatic;
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.protobuf.util.Timestamps;
+import io.cdap.cdap.securestore.spi.secret.SecretMetadata;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Wrapper around a {@link io.cdap.cdap.securestore.spi.secret.SecretMetadata} which allows it to
+ * easily be converted a GCP SecretManager {@link com.google.cloud.secretmanager.v1.Secret} and
+ * back.
+ *
+ * <p>Does not contain the underlying secret payload.
+ */
+public final class WrappedSecret {
+  private static final Gson gson = new Gson();
+
+  private final String namespace;
+  private final SecretMetadata secretMetadata;
+
+  private WrappedSecret(String namespace, SecretMetadata secretMetadata) {
+    this.namespace = namespace;
+    this.secretMetadata = secretMetadata;
+  }
+
+  /** Constructs a new WrappedSecret from a CDAP Secret. */
+  public static WrappedSecret fromMetadata(String namespace, SecretMetadata metadata) {
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  /** Constructs a new WrappedSecret from a GCP Secret Manager Secret. */
+  public static WrappedSecret fromGcpSecret(Secret secret) throws IOException {
+    String namespace = getNamespace(secret);
+    SecretMetadata metadata = toSecretMetadata(secret);
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+  public SecretMetadata getCdapSecretMetadata() {
+    return secretMetadata;
+  }
+
+  /**
+   * Returns a new GCP {@link Secret} representing the underlying SecretMetadata.
+   *
+   * @param resourceName Value to set for the "name" field needed for update operations.
+   */
+  public Secret getGcpSecret(String resourceName) {
+    return Secret.newBuilder()
+        // Set replication policy to automatic (as opposed to user-managed) and do not specify a
+        // CMEK (use google-managed key).
+        .setReplication(Replication.newBuilder().setAutomatic(Automatic.getDefaultInstance()))
```

Comment:
style: the rest of cdap uses 2 space indentation after a newline

Reply:
commented elsewhere but I tried to do this. 

Too spoiled by google-java-format making it so I don't have to think about formatting.